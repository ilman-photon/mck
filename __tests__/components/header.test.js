const ReactTestRenderer = require("react-test-renderer");
import HeaderComponent from "@/components/header";

jest.mock("next/router", () => {
  return {
    useRouter: jest.fn(),
  };
});

it("renders correctly", () => {
  const tree = ReactTestRenderer.create(
    <HeaderComponent>McKesson</HeaderComponent>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
