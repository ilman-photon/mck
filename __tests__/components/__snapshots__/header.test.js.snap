// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header flex lg:grid container sticky mx-auto blue-txt bg-mckwhite border-b border-mckblue"
  id="header"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="flex"
    style={{}}
  >
    <div
      className="hamburger-menu lg:hidden xl:hidden"
      onClick={[Function]}
    >
      <div
        className="bar "
      />
    </div>
  </div>
  <div
    className="brand-logo"
    onClick={[Function]}
  >
    <img
      alt="Foster Thrive logo"
      className="lg:mt-7"
      id="logo-image"
      src="images/logo.png"
      tabIndex={0}
      style={{}}
    />
  </div>
  <div
    className="lg:w-full flex border-0 lg:border-b border-mcknormalgrey w-18 header-sticky"
    style={{}}
  >
    <div
      className="md:flex container lg:flex w-8/12 ml-auto mobile-customenav "
      id="nav-bar"
    >
      <div
        className="lg:mx-auto lg:flex xl:mx-auto xl:flex isMobileUi"
      />
    </div>
    <div
      className="relative w-64 search-box"
    >
      <button
        aria-label="search-icon"
        aria-describedby="clikable search item listed"
        className="btn-search w-12 h-12 bg-transparent right-4 absolute cursor-pointer outline-none font-black border-0"
      />
      <input
        aria-label="search-bar"
        className="input-search outline-0 h-12 w-12 border-none text-sofia-reg placeholder:text-sofia-reg text-base placeholder:text-base font-normal placeholder:font-normal tracking-normal placeholder:tracking-normal rounded px-3 py-4 pr-10 bg-transparent absolute inset-y-0 right-3.5 blue-txt focus:w-80 focus:bg-white focus:top-1   "
        onKeyDown={[Function]}
        placeholder="Search..."
      />
      <svg
        aria-label="search-icon"
        className="search-icon absolute lg:top-4 -top-1 lg:right-6 right-2 "
        height="19"
        role="img"
        viewBox="0 0 18 19"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.7549 11.734H11.9649L11.6849 11.464C12.6649 10.324 13.2549 8.84401 13.2549 7.23401C13.2549 3.64401 10.3449 0.734009 6.75488 0.734009C3.16488 0.734009 0.254883 3.64401 0.254883 7.23401C0.254883 10.824 3.16488 13.734 6.75488 13.734C8.36488 13.734 9.84488 13.144 10.9849 12.164L11.2549 12.444V13.234L16.2549 18.224L17.7449 16.734L12.7549 11.734ZM6.75488 11.734C4.26488 11.734 2.25488 9.72401 2.25488 7.23401C2.25488 4.74401 4.26488 2.73401 6.75488 2.73401C9.24488 2.73401 11.2549 4.74401 11.2549 7.23401C11.2549 9.72401 9.24488 11.734 6.75488 11.734Z"
          fill="#001A71"
        />
      </svg>
    </div>
  </div>
</div>
`;
